src/tools/wasm-link.cc

```diff
@@ -68,53 +53,26 @@ struct Context {
   ssize_t current_section_payload_offset = 0;
 };
 
-static void on_option(struct OptionParser* parser,
-                      struct Option* option,
-                      const char* argument) {
-  switch (option->id) {
-    case FLAG_DEBUG:
-      s_debug = true;
-      s_log_stream = FileStream::CreateStdout();
-      break;
-
-    case FLAG_OUTPUT:
-      s_outfile = argument;
-      break;
-
-    case FLAG_RELOCATABLE:
-      s_relocatable = true;
-      break;
-
-    case FLAG_HELP:
-      print_help(parser, PROGRAM_NAME);
-      exit(0);
-      break;
```

Comment:
You're right, it is a bit noisy. I was a bit concerned about that, so I tried to add helper functions to make the common cases nice. One additional thought I had was to have an explicit BoolOption/StringOption so you could just do:
```
parser.AddBoolOption('f', "fold-exprs", "...", &s_fold_exprs_value);
```

But I figure that can be layered on top of this change, which is more general.